File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        maxlen = 0
        sy = set()
        l =0 

        for i in range(len(s)):
            while s[i] in sy:
                sy.remove(s[l])
                l += 1
            sy.add(s[i])

            maxlen = max(maxlen,i-l+1)
        return maxlen
        

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) 
    {
        int l = 0;
        int len = 0;
        int n = s.size();
        unordered_set<char> set;

        for(int i = 0 ; i < n ;i++)
        {
            while (set.contains(s[i]))
            {
                set.erase(s[l]);
                l++;
            }
            set.insert(s[i]);
            len = max(len,i-l+1);
        }
        return len;

        
    }
};